sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/ConfigurationManager.cs

```diff
@@ -2,77 +2,179 @@
+
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new MissingFieldException(
+                        string.Format(
+                            provider: CultureInfo.CurrentCulture,
+                            format: AuthenticationEventResource.Ex_Trigger_AuthorityUrl_Required,
+                            arg0: AuthorityUrlKey));
+                }
+
+                return value;
             }
-            else if (SERVICES.ContainsKey(serviceId))
+        }
+
+        /// <summary>
+        /// Get the OpenId connection host from the environment variable or use the default value.
+        /// OPTIONAL FEILD, defaults to public cloud id.
+        /// </summary>
+        internal string AuthorizedPartyAppId
+        {
+            get
             {
-                serviceInfo = SERVICES[serviceId];
+                string value = GetConfigValue(AuthorizedPartyAppIdKey, triggerAttribute?.AuthorizedPartyAppId);
+
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new MissingFieldException(
+                        string.Format(
+                            provider: CultureInfo.CurrentCulture,
+                            format: AuthenticationEventResource.Ex_Trigger_AuthorizedPartyApplicationId_Required,
+                            arg0: AuthorizedPartyAppIdKey));
+                }
+
+                return value;
             }
+        }
+
+        /// <summary>
+        /// If we should bypass the token validation.
+        /// Use only for testing and development.
+        /// </summary>
+        internal bool BypassValidation => GetConfigValue(BYPASS_VALIDATION_KEY, false);
 
-            return serviceInfo != null;
+        /// <summary>
+        /// If the EZAuth is enabled.
+        /// </summary>
+        internal bool EZAuthEnabled => GetConfigValue(EZAUTH_ENABLED, false);
+
+        /// <summary>
+        /// Get the issuer string based on the token schema version.
+        /// </summary>
+        /// <param name="tokenSchemaVersion">v2 will return v2 odic url, v1 will return v1</param>
+        /// <returns></returns>
+        internal string GetOpenIDConfigurationUrlString(SupportedTokenSchemaVersions tokenSchemaVersion)
+        {
+            return tokenSchemaVersion == SupportedTokenSchemaVersions.V2_0 ?
+                AuthorityUrl + OpenIdConfigurationPathV2 :
+                AuthorityUrl + OpenIdConfigurationPath;
+        }
+
+        /// <summary>
+        /// Validate the authorization party is accurate to the one in configuration.
+        /// </summary>
+        /// <param name="authoizedPartyValueFromTokenOrHeader">The value from either the token or the header.</param>
+        /// <returns>True if azp/appid value matches the configured value.</returns>
+        internal bool ValidateAuthorizationParty(string authoizedPartyValueFromTokenOrHeader)
```

Comment:
I don't think this logic belongs in config manager. this is purely a validation logic

Reply:
Moved to helper